Mathlib/Algebra/Order/Group/Cone.lean

```diff
@@ -1,85 +1,131 @@
 /-
 Copyright (c) 2017 Mario Carneiro. All rights reserved.
 Released under Apache 2.0 license as described in the file LICENSE.
-Authors: Mario Carneiro, Scott Morrison
+Authors: Mario Carneiro, Scott Morrison, Artie Khovanov
 -/
 import Mathlib.Algebra.Order.Group.Defs
+import Mathlib.Algebra.Order.Monoid.Submonoid
 
 /-!
 # Construct ordered groups from positive cones
 
 In this file we provide structures `PositiveCone` and `TotalPositiveCone`
 that encode axioms of `OrderedAddCommGroup` and `LinearOrderedAddCommGroup`
-in terms of the `(0 ≤ ·)` predicate.
+in terms of the subset of non-negative elements.
 
 We also provide two constructors,
 `OrderedAddCommGroup.mkOfPositiveCone` and `LinearOrderedAddCommGroup.mkOfPositiveCone`,
 that turn these structures into instances of the corresponding typeclasses.
 -/
 
+variable {S G : Type*} [AddCommGroup G] [SetLike S G] (C : S)
+
 namespace AddCommGroup
 
-/-- A collection of elements in an `AddCommGroup` designated as "non-negative".
-This is useful for constructing an `OrderedAddCommGroup`
-by choosing a positive cone in an existing `AddCommGroup`. -/
--- Porting note(#5171): @[nolint has_nonempty_instance]
-structure PositiveCone (α : Type*) [AddCommGroup α] where
-  /-- The characteristic predicate of a positive cone. `nonneg a` means that `0 ≤ a` according to
-  the cone. -/
-  nonneg : α → Prop
-  /-- The characteristic predicate of a positive cone. `pos a` means that `0 < a` according to
-  the cone. -/
-  pos : α → Prop := fun a => nonneg a ∧ ¬nonneg (-a)
-  pos_iff : ∀ a, pos a ↔ nonneg a ∧ ¬nonneg (-a) := by intros; rfl
-  zero_nonneg : nonneg 0
-  add_nonneg : ∀ {a b}, nonneg a → nonneg b → nonneg (a + b)
-  nonneg_antisymm : ∀ {a}, nonneg a → nonneg (-a) → a = 0
-
-/-- A positive cone in an `AddCommGroup` induces a linear order if
-for every `a`, either `a` or `-a` is non-negative. -/
--- Porting note(#5171): @[nolint has_nonempty_instance]
-structure TotalPositiveCone (α : Type*) [AddCommGroup α] extends PositiveCone α where
-  /-- For any `a` the proposition `nonneg a` is decidable -/
-  nonnegDecidable : DecidablePred nonneg
-  /-- Either `a` or `-a` is `nonneg` -/
-  nonneg_total : ∀ a : α, nonneg a ∨ nonneg (-a)
-
-/-- Forget that a `TotalPositiveCone` is total. -/
-add_decl_doc TotalPositiveCone.toPositiveCone
-
-end AddCommGroup
-
-namespace OrderedAddCommGroup
+/-- `PositiveConeClass S G` says that `S` is a type of `PositiveCone`s in `G`. -/
```

Comment:
```suggestion
/-- `PositiveConeClass S G` says that `S` is a type of positive cones in `G`. -/
```
to avoid confusing the mathlib type with the mathematical notion